### PR TITLE
Let the user select the type of derivative computation in the Deriv class

### PR DESCRIPTION
**Pull request by fevangelista**
## Description
This PR introduces an option in the call to `Deriv::compute()` so that the user can specify the type of gradient computation. Currently, this class determines which procedure to follow using an internal logic that is not sufficiently flexible to accommodate plugins.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Add deric_calc_type option to `Deriv::compute()`
- [x] Define enum and python interface

## Checklist
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/export_mints.cc [hunk 1]
```diff
@@ -644,6 +646,12 @@ void export_mints(py::module& m) {
           "Returns the multiplication of three matrics A, B, and C, with options to transpose each beforehand", "A"_a,
           "B"_a, "C"_a, "transA"_a = false, "transB"_a = false, "transC"_a = false);
 
+    py::enum_<DerivCalcType>(m, "DerivCalcType")
+        .value("Default", DerivCalcType::Default)
+        .value("SCF", DerivCalcType::SCF)
+        .value("SCFandDF", DerivCalcType::SCFandDF)
+        .value("Correlated", DerivCalcType::Correlated);
```
Comment (loriab) [inline psi4/src/export_mints.cc]: fyi, starting fairly recently, one can finally add docstrings to enums in pb11. https://github.com/loriab/pylibefp/blob/master/pylibefp/core.cc#L493

**Comment by JonathonMisiewicz** [comment]
I mentioned this on Slack, but at present, only the Correlated procedure is used in Psi, and the others are legacy code. I have no idea what plugins are doing, but I assumed those all used Correlated as well. Let me know if you're going to change that. I imagine that you're using the SCF procedure for CASSCF purposes?

**Comment by fevangelista** [comment]
We need the correlated procedure to run CASSCF and  may adapt the code to use DF once the new Deriv code is available. In putting together this PR, my intention was to make very limited changes and only improve the functionality of the current code. Since I am not sure what plugins do, I just reproduced the old functionality as it was before. I'd be happy to delete some of the legacy code if we are certain it is not used.

File: psi4/src/export_mints.cc (hunk 1 above)
Comment (fevangelista) [reply psi4/src/export_mints.cc]: Done.

**Comment by JonathonMisiewicz** [comment]
I don't think we _can_ be certain what `Psi_API` code is being used and what isn't... We probably should leave it for now, with the possibility of deprecating that old system entirely when I get the new one working properly.

**Comment by fevangelista** [comment]
@loriab: I indeed forgot to pass the default to python. Pushed a commit and waiting for testing. In the long run the default should disappear.

**Pull request merged**